### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by sandhu**

### CODE REVIEW

File: src/leiningen/new/web_7theta/src/ring_handler.clj [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   (routes
+    (create-resource-handler {:path "/"})
+    (create-default-handler)))){{#graal?}}
+
+;; boilerplate for graal
+(defmethod response/resource-data :resource
+  [^java.net.URL url]
+  (let [conn (.openConnection url)]
+    {:content (.getInputStream conn)
+     :content-length (let [len (.getContentLength conn)] (if-not (pos? len) len))})){{/graal?}}
```
Comment (joel7t) [inline src/leiningen/new/web_7theta/src/ring_handler.clj:26]: Convention: Is there a missing else-clause, or should this be a `when-not`?

Also, the logic is unexpected: only put in the content length if it's 0 or negative?

File: src/leiningen/new/web_7theta/src/views.cljs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+(ns {{ns-name}}.views
+    (:require [{{name}}.macros :refer [defnc]]
+              [helix.core :refer [$]]
+              [helix.dom :refer [div button]]
+              [tailwind.core :refer [tw]]
+              [via.core :refer [subscribe dispatch invoke]]
+              [signum.hooks :refer [use-signal]]{{#routing?}}
+              [{{ns-name}}.modules :as modules]{{/routing?}}))
```
Comment (joel7t) [inline src/leiningen/new/web_7theta/src/views.cljs:8]: We had talked about conventions in the listing of `:requires`s.  Should `{{ns-name}}.modules` be the second require?

File: src/leiningen/new/web_7theta/prod/main.clj [hunk 3]
```diff
@@ -0,0 +1,16 @@
+(ns {{name}}.main
+    (:require [{{name}}.config :refer [config]]
```
Comment (joel7t) [inline src/leiningen/new/web_7theta/prod/main.clj:2]: `ns-name` instead of `name`?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# web-7theta
+
+[![Current Version](https://img.shields.io/clojars/v/web-7theta/lein-template.svg)](https://clojars.org/web-7theta/lein-template)
```
Comment (tom7t) [inline README.md:3]: Is this url correct? `web-7theta` and `lein-template` ?

File: src/leiningen/new/web_7theta.clj [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+(defn web-7theta
+  [name & options]
+  (check-options options)
+  (let [options (->> options
+                     (filter (partial available-options))
+                     (map #(keyword (apply str (rest %))))
+                     set)
+        data (template-data name options)]
+    (main/info "Generating web project with 7theta template.")
```
Comment (tom7t) [inline src/leiningen/new/web_7theta.clj:34]: Should be `... 7theta web template.` I think.

File: src/leiningen/new/web_7theta/prod/main.clj [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+(defn -main [& args]
+  (let [env (System/getenv)
+        port (or ((fsafe string->long) (get env "PORT")) 8080)
+        host (or (get env "HOST") "0.0.0.0")
+        app (-> config
+                (assoc-in [:via/http-server :host] host)
+                (assoc-in [:via/http-server :port] port)
+                ig/init)]
+    (println "URL:" (str "http" "://" host ":" port))))
```
Comment (tom7t) [inline src/leiningen/new/web_7theta/prod/main.clj:16]: One issue is that on windows, browsing to `http://0.0.0.0:3449` doesn't work. Should we check for `0.0.0.0` and print out `localhost` for the host instead, for compatibility with our electron work in structor?

File: README.md (hunk 4 above)
Comment (sandhu) [reply README.md:3]: The URLs are correct. It has to do with how lein templates get resolved.

File: src/leiningen/new/web_7theta/src/ring_handler.clj [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  (ring-handler
+   (router
+    [[default-via-endpoint via-handler]
+     ["/" (fn [_]
+            (-> "public/index.html"
+                response/resource-response
+                (response/content-type "text/html")))]])
+   (routes
+    (create-resource-handler {:path "/"})
+    (create-default-handler)))){{#graal?}}
```
[the remark this answers is not in the document]
Comment (sandhu) [reply src/leiningen/new/web_7theta/src/ring_handler.clj:19]: For an unknown resource we should return a 404. For the client side routing, I assume we'd use the part after the `#`

[the remark this answers is not in the document]
Comment (sandhu) [reply src/leiningen/new/web_7theta/src/ring_handler.clj:19]: This makes sense when `+routing` is present adn th

**Review by joel7t — APPROVED** [review]

**Pull request merged**